feat: backstage using devcontainer and Daytona

anuragkainth opened the pull request:
- Configured backstage app using app-config.yaml file.
- Added github auth provider configs as described in readme.md
- Added devcontainer.json configuration for smooth Backstage workspace setup using Daytona.

anuragkainth committed:
chore: devcontainer added

anuragkainth committed:
chore: backstage intial commit

anuragkainth committed:
chore: app-config updated

Tpuljak commented on README.md:
```diff
@@ -162,7 +162,7 @@ This configuration includes:
 - **overrideFeatureInstallOrder**: Specifies the order of feature installation to ensure common utilities, Git, Node.js, and TypeScript are set up in the correct sequence.
 - **customizations**: Installs essential Visual Studio Code extensions, including Prettier, ESLint, a spell checker, and TypeScript Next.
-- **workspaceFolder**: Sets the workspace folder to `/workspaces/starter-backstage` to match the local workspace folder name.
+- **workspaceFolder**: Sets the workspace folder to `/workspaces/${localWorkspaceFolderBasename}` to match the local workspace folder name.
```
`"workspaceFolder": "/workspaces/daytona-backstage-starter"` this is set in the `devcontainer.json` file

anuragkainth replied:
Thankyou for pointing that out. I have updated both `devcontainer.json` and `readme.md` for this branch.

anuragkainth committed:
fix: workspace folder name corrected

anuragkainth committed:
docs: readme updated

anuragkainth committed:
docs: env setup steps updates

anuragkainth committed:
Merge branch 'main' into feat/first-iteration

nibzard commented on README.md:
````diff
@@ -26,7 +26,29 @@ This repository contains a Backstage application set up within a Dev Container e
+```
+2. **Create Daytona Workspace:**
+```bash
+daytona create https://github.com/daytonaio-experiments/starter-backstage.git -t local
````
why -t local

anuragkainth replied:
we are targeting local machine using `-t local`. But removing it will not cause any issue. Should we remove it next commit ?

nibzard commented on README.md:
````diff
@@ -71,27 +93,7 @@ Now that we are allowing users to sign in to Backstage, we should allow them to
 ```bash
 export GITHUB_TOKEN=<PAT_TOKEN>
````
.env.example is missing, and export is adding to env instead of .env file?

anuragkainth replied:
adding .env.example file. we have already insured that these variables are available to backstage during run time in order for them to get accessed in app-config.yaml through script in package.json. I believe we can remove `export` from here.

nibzard commented on README.md:
```diff
@@ -61,7 +83,7 @@ catalog:
       orgs: ['<Your_Organization_Name>']
```
is this optional as above it seems optional?

anuragkainth replied:
This would be necessary if we want members of our github organization to access the backstage for collaboration.

anuragkainth committed:
fix: refactor and cleanup

anuragkainth commented:
- added .env.example file for user reference.
- removed `export` from .env section of readme file.
- removed `-t local` from readme and compressed daytona commands using `--code` flag.
- code cleanup for app-config.json file.

anuragkainth committed:
updated readme.md
